extensions/wikia/AdMixExperiment/scripts/AdMixExperiment.js

```diff
@@ -0,0 +1,340 @@
+			} else {
+				adAndRecircWrapperStyles.top = (stopPosition - topAdHeight - globalNavigationHeight + viewportHeight - adMixRecircWrapperHeight - noRecircAdBottomSpace) + 'px';
+			}
+
+			$adAndRecircWrapper.css(adAndRecircWrapperStyles);
+		}
+
+		function resetRecircStyles() {
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', (viewportWidth > breakpointSmall ? gapSize : 0) + 'px');
+
+				$recirc.css({
+					position: '',
+					top: '',
+					width: ''
+				});
+			}
+
+			resetAdAndRecircWrapper();
+		}
+
+		function resetTopAd() {
+			$topRightAd.css({
+				position: '',
+				top: '',
+				width: ''
+			});
+		}
+
+		function resetTopAdPaddings() {
+			$topRightAdWrapper.css({
+				'padding-top': '',
+				'padding-bottom': gapSize + 'px'
+			});
+		}
+
+		function resetAdAndRecircWrapper() {
+			$adAndRecircWrapper.css({
+				bottom: '',
+				position: '',
+				top: '',
+				width: ''
+			});
+		}
+
+		function update() {
+			var scrollTop = $window.scrollTop();
+			var firstAdTopPosition = getFirstAdTopPosition();
+			var firstAdBottomPosition;
+			var adAndRecircWrapperTopPosition = getAdAndRecircWrapperTopPosition();
+			var adAndRecircWrapperBottomPosition;
+			var stopPosition;
+
+			if (
+				scrollTop < firstAdTopPosition ||
+				(!topRightAdFixed && scrollTop < adAndRecircWrapperTopPosition) ||
+				viewportWidth <= breakpointSmall
+			) {
+				reset();
+				return;
+			}
+
+			firstAdBottomPosition = getFirstAdBottomPosition(firstAdTopPosition);
+
+			if (
+				scrollTop >= firstAdTopPosition &&
+				scrollTop < firstAdBottomPosition
+			) {
+				scrollAfterFirstAdTopPosition();
+				return;
+			}
+
+			if (
+				scrollTop >= firstAdBottomPosition &&
+				scrollTop < adAndRecircWrapperTopPosition
+			) {
+				scrollAfterFirstAdBottomPosition();
+				return;
+			}
+
+			adAndRecircWrapperBottomPosition = getAdAndRecircWrapperBottomPosition(adAndRecircWrapperTopPosition);
+
+			if (
+				scrollTop >= adAndRecircWrapperTopPosition &&
+				scrollTop < adAndRecircWrapperBottomPosition
+			) {
+				scrollAfterAdAndRecircWrapperTopPosition();
+				return;
+			}
+
+			adMixRecircWrapperHeight = $adAndRecircWrapper.outerHeight(true);
+			stopPosition = getStopPosition();
+
+			if (
+				scrollTop >= adAndRecircWrapperBottomPosition &&
+				scrollTop < stopPosition
+			) {
+				scrollAfterAdAndRecircWrapperBottomPosition();
+			} else if (scrollTop >= stopPosition) {
+				scrollStopPosition(stopPosition);
+			}
+		}
+
+		function onRightRailReady() {
+			$recircWrapper = $('#recirculation-rail');
+			$adAndRecircWrapper = $('#WikiaAdInContentPlaceHolder');
+
+			if (recircEnabled) {
+				$adAndRecircWrapper.addClass('ad-mix-experiment-enabled');
+			}
+
+			// There is no recirculation right rail module on non-en wikis
+			if (window.wgContentLanguage === 'en') {
+				$recircWrapper.one('premiumRecirculationRail.ready', onRecircReady);
+			} else {
+				// TODO check which experiment needs it
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)